### PR TITLE
Greatly simplify the ImageFetcher and tests

### PR DESCRIPTION
**Pull request by cjsauer**
All the future code can basically be replaced by Meteor.wrapAsync, which takes a function that requires a callback, and converts it into a synchronous function.

I also removed the getUrls module function, which just seemed redundant when we already are npmRequiring the google-images package.

### CODE REVIEW

**Comment by marti584** [comment]
Wow. This is a rather large change. So just wrapping the search call in wrapAsync() makes the callback run synchronously? I guess I was confused about how to implement that when I was first going over how to create the module. It looks good though :+1:

**Comment by cjsauer** [comment]
Right. Check out the [docs for wrapAsync](http://docs.meteor.com/#/full/meteor_wrapasync). Behind the curtain, Meteor handles all that Promise related stuff, and will pause the fiber automatically (on the server). On the client, you still need a callback, otherwise you'd hang the browser.

**Comment by cjsauer** [comment]
I'm also no longer exporting that google-images package as a global module. I think that's safer. Now we only export that one search function *and* make it synchronous. So now when you call `Modules.server.googleImageSearchSync` you know exactly what you're getting.

**Comment by cjsauer** [comment]
:+1:

File: server/modules/ImageFetcher.js [hunk 1]
```diff
@@ -1,29 +1,2 @@
...
-    },
-    page: page
-  };
-  var t = Modules.server.GoogleImageSearcher.search(channel.get('query'), options);
-  return fut.wait();
-};
-
-Modules.server.getUrls = getUrls;
+GoogleImageSearcher = Meteor.npmRequire('google-images');
+Modules.server.googleImageSearchSync = Meteor.wrapAsync(GoogleImageSearcher.search);
```
Comment (elGatoMantocko) [inline server/modules/ImageFetcher.js]: Wow. So sex.

**Comment by elGatoMantocko** [comment]
:+1:

**Pull request merged**